source/agora/test/Base.d

```diff
@@ -710,7 +710,14 @@ public class TestAPIManager
         clients_idxs.each!(idx =>
             enrolls.enumerate.each!((idx_enroll, enroll) {
                 if (clients_idxs.canFind(idx_enroll))
-                    retryFor(this.clients[idx].getPreimage(enroll.utxo_key).height >= height,
+                    retryFor(
+                        ()
+                        {
+                            Set!Hash enroll_keys;
+                            enroll_keys.put(enroll.utxo_key);
+                            auto preimages = this.clients[idx].getPreimagesForEnrollKeys(enroll_keys);
```

Comment:
You could just use `[enroll.utxo_key]` if it was a `Hash`.

Reply:
`Set!Hash.from(enroll.utxo_key.only)` should do it ([`only`](https://dlang.org/phobos/std_range.html#only) is in `std.range`).